src/app/fdctl/run/tiles/fd_replay.c

```diff
@@ -466,41 +483,37 @@ unprivileged_init( fd_topo_t *      topo,
     FD_LOG_ERR(( "no blocktore workspace" ));
   }
 
-  ctx->last_stake_weights_push_time = 0;
+  /* Funk setup */
+  ulong funk_obj_id = fd_pod_queryf_ulong( topo->props, ULONG_MAX, "funk" );
+  FD_TEST( funk_obj_id!=ULONG_MAX );
+  ctx->funk_wksp = topo->workspaces[ topo->objs[ funk_obj_id ].wksp_id ].wksp;
 
-  /* Valloc setup */
-  void * alloc_shalloc = fd_alloc_new( alloc_shmem, 3UL );
-  if( FD_UNLIKELY( !alloc_shalloc ) ) { 
-    FD_LOG_ERR( ( "fd_allow_new failed" ) ); }
-  fd_alloc_t * alloc = fd_alloc_join( alloc_shalloc, 3UL );
-  if( FD_UNLIKELY( !alloc ) ) {
-    FD_LOG_ERR( ( "fd_alloc_join failed" ) ); 
+  if( ctx->funk_wksp == NULL ) {
+    FD_LOG_ERR(( "no funk workspace" ));
   }
 
-  char hostname[64];
-  gethostname(hostname, sizeof(hostname));
-  ulong hashseed = fd_hash(0, hostname, strnlen(hostname, sizeof(hostname)));
-
-  // Allocate new wksp
-  fd_wksp_t * funk_wksp = fd_wksp_new_anonymous( FD_SHMEM_GIGANTIC_PAGE_SZ, 8UL, 0UL, "wksp", 0UL );
-  if (funk_wksp == NULL)
-    FD_LOG_ERR(( "failed to attach to workspace" ));
-  fd_wksp_reset( funk_wksp, (uint)hashseed);
-
-  fd_scratch_attach( smem, fmem, SCRATCH_MAX, SCRATCH_DEPTH );
-
-  // Funky setup
   fd_funk_t * funk;
   void * shmem;
-  shmem = fd_wksp_alloc_laddr( funk_wksp, fd_funk_align(), fd_funk_footprint(), FD_FUNK_MAGIC );
+  shmem = fd_wksp_alloc_laddr( ctx->funk_wksp, fd_funk_align(), fd_funk_footprint(), FD_FUNK_MAGIC );
   if (shmem == NULL)
     FD_LOG_ERR(( "failed to allocate a funky" ));
-  funk = fd_funk_join(fd_funk_new(shmem, 42, hashseed, 1024UL, 300000UL));
+  funk = fd_funk_join( fd_funk_new( shmem, FD_FUNK_MAGIC, ctx->funk_seed, tile->replay.txn_max, tile->replay.index_max ) );
```

Comment:
I remember a conversation with michael about having these set up with the topology. Is that a future change (or not relevant/ignored)

Reply:
this is doing that now